From 6d0637df0c2ae0bc2ac2d0b7ba24f7c302fd5baf Mon Sep 17 00:00:00 2001
From: Haitao Huang <haitao.huang@intel.com>
Date: Mon, 13 Aug 2012 11:06:41 -0500
Subject: stagefright: add support for the quirk kRequiresLoadedToIdleAfterAllocation.

This is needed for meta data buffer mode because Google configures meta data mode
after configuration stage, while it should be during configuration to be strictly
OMX compliant.

For some encoders such as Intel hw AVC encoder implementation, this has to be
done during configuration for it configure/init pipelines in separate thread thus
can't switch to meta buffer mode later after configuration.

Change-Id: I84d8db8639f6ce1cacd050f0cdf44dd0a1b73816
Signed-off-by: Haitao Huang <haitao.huang@intel.com>
---
 media/libstagefright/OMXCodec.cpp |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/OMXCodec.cpp b/media/libstagefright/OMXCodec.cpp
index 9d349a1..4f38ffd 100644
--- a/media/libstagefright/OMXCodec.cpp
+++ b/media/libstagefright/OMXCodec.cpp
@@ -248,6 +248,10 @@ uint32_t OMXCodec::getComponentQuirks(
                 index, "output-buffers-are-unreadable")) {
         quirks |= kOutputBuffersAreUnreadable;
     }
+    if (list->codecHasQuirk(
+                index, "requires-loaded-to-idle-after-allocation")) {
+        quirks |= kRequiresLoadedToIdleAfterAllocation;
+    }
 
     return quirks;
 }
-- 
1.7.4.1

